game/world/managers/objects/units/player/StatManager.py

```diff
@@ -415,6 +417,19 @@ def update_base_mana_regen(self):
         regen = CLASS_BASE_REGEN_MANA[player_class] + spirit * CLASS_SPIRIT_SCALING_MANA[player_class]
         self.base_stats[UnitStats.POWER_REGENERATION_PER_5] = int(regen / 2)
 
+    def update_base_melee_critical_chance(self):
+        if self.unit_mgr.get_type_id() != ObjectTypeIds.ID_PLAYER:
+            return
+
+        player_class = self.unit_mgr.player.class_
+        agility = self.get_total_stat(UnitStats.AGILITY)
```

Comment:
In 0.5.3 Agility doesn't increase critical chance, it's Strength what increases it (you can confirm it by checking the description of each stat in game).